sqladmin/_queries.py

```diff
@@ -45,7 +45,12 @@ def _set_attributes_sync(self, session: Session, obj: Any, data: dict) -> Any:
 
             if not value:
                 # Set falsy values to None, if column is Nullable
-                if not relation and column.nullable and value is not False:
+                if (
+                    not relation
+                    and column.nullable
+                    and value is not False
```

Comment:
Not really sure, but could use `bool(value)` to handle all falsy cases.
The downside is that there might be some types with unexpected behaviour.

Reply:
bool(None) still returns False instead of None making it indistinguishable from something that is false or zero and there are some variations in different handling by versions of python (e.g. empty arrays/mappings). Probably better to handle known cases explicitly to ensure compatibility across python versions. But I would be happy with any solution that allows me to set an integer field to 0 :)

Reply:
Yeah probably you are right.
After thinking about this more and looking into the history, I think we can do something like:

```py
if not value:
                # Set falsy values to None, if column is Nullable
                if (
                    not relation
                    and column.nullable
                    and isinstance(value, bool)
                    and value is not False
                ):
                    value = None

                setattr(obj, key, value)
                continue
```

So we don't need to handle every type, the case that needs to be handled is to distinguish nullable booleans, otherwise we can set whatever value avaialble to the column.
And yes this works in your case of zero :)